File: 2025-03-March-GFG-POTD/March 28 - Activity Selection/Solution.py
class Solution:
    def activitySelection(self, start, finish):
        activities = sorted(zip(finish, start))  # Sort activities by finish time
        count = 1
        last_end_time = activities[0][0]

        for i in range(1, len(activities)):
            if activities[i][1] > last_end_time:
                count += 1
                last_end_time = activities[i][0]

        return count

if __name__ == "__main__":
    t = int(input().strip())
    for _ in range(t):
        start = list(map(int, input().strip().split()))
        finish = list(map(int, input().strip().split()))

        obj = Solution()
        print(obj.activitySelection(start, finish))
        print("~")

File: 2025-03-March-GFG-POTD/March 28 - Activity Selection/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int activitySelection(vector<int>& start, vector<int>& finish) {
        int n = start.size();
        vector<pair<int, int>> activities;

        for (int i = 0; i < n; i++) {
            activities.push_back({finish[i], start[i]});
        }

        sort(activities.begin(), activities.end());

        int count = 1;
        int lastEndTime = activities[0].first;

        for (int i = 1; i < n; i++) {
            if (activities[i].second > lastEndTime) {
                count++;
                lastEndTime = activities[i].first;
            }
        }

        return count;
    }
};

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> start(n), finish(n);
        for (int i = 0; i < n; i++) cin >> start[i];
        for (int i = 0; i < n; i++) cin >> finish[i];

        Solution obj;
        cout << obj.activitySelection(start, finish) << endl;
        cout << "~" << endl;
    }
    return 0;
}
